Automattic/wp-calypso
# Jetpack Settings: Introduce an updated custom content types card

### Purpose

This PR introduces an updated version of the Custom Content Types card. The old card was no longer matching the rest of the tab, since it was using the large toggles and its layout was quite different. This PR is part of #9171.

### What this PR does

* Introduces a new Custom Content Types card that works and looks just like the other cards in the same tab do. This is also outlined in p6TEKc-HQ-p2.
* Uses `wrapSettingsForm` and the rest of the already laid out infrastructure to implement the new card.
* Removes a ton of boilerplate code from the writing form, as well as the legacy card component.

### Preview

**The old Custom Content Types card**
![](https://cldup.com/1bBdTVxcLw.png)

**The new Custom Content Types card with enabled module**
![](https://cldup.com/n7P2sPZPh1.png)

**The new Custom Content Types card with disabled module**
![](https://cldup.com/DZfeO_lckd.png)

### Notes

This PR includes #11095, which improves tests for the `filterSettingsByActiveModules()` methods.

### To test

* Checkout this branch
* Go to `/settings/writing/$site` where `$site` is a Jetpack site.
* Play around with the module toggle, enable and disable it and verify it works as expected.
* Play around with the Testimonials and Portfolio toggles and the Save Changes button and verify everything saves and loads correctly.
* Verify the card is not shown for Jetpack < 4.5
* Verify the card is not shown for WordPress.com sites.

/cc
@oskosk @roccotripaldi @johnHackworth @ryelle for code review
@MichaelArestad @rickybanister for design review
